test/require/require.md

```diff
@@ -0,0 +1,17 @@
+# Using local library
+
+```ocaml
+# #require "example_lib";;
```

Comment:
I would tend to favour the removal of the `#require` statements as they: 1/ force a dependency towards ocamlfind and 2/ could be guessed properly in most of the cases (see utop or [down](https://github.com/dbuenzli/down)). 

So I think it's better to only use `require=...` in the block metadata to indicate which OCaml library should be added to the scope of the current environment and remove all `#require` completely.

Reply:
There is a lot of `#require` in rwo, that's why we added that. In duniverse mode, they are used to specify package deps, otherwise they do nothing.

Reply:
are `#require` used elsewhere than in preludes in rwo?

Reply:
I think `#require` statement should appear in toplevel code blocks so that the reader can eventually copy and paste the content of those blocks into a toplevel and get it to work.

There are indeed quite a few `#require` statement in code blocks in RWO.

Also note that parsing them doesn't necessarily mean that they have to be evaluated, to the contrary, we could strip them from the actual lines passed on to the toplevel and load the libraries there ourselves.

Reply:
IThere are not so many `#require` statements in RWO so these can be changed if needed:
```
$ git grep "#require" book/**/*.md | grep -v prelude
book/concurrent-programming/README.md:# #require "async"
book/concurrent-programming/README.md:We first load the Async package in the toplevel using `#require`, and then
book/concurrent-programming/README.md:# #require "cohttp.async"
book/foreign-function-interface/README.md:# #require "ctypes.foreign"
book/foreign-function-interface/README.md:# #require "ctypes.top"
book/json/README.md:# #require "yojson"
book/lists-and-patterns/README.md:# #require "core_bench"
book/records/README.md:# #require "re.posix"
book/runtime-memory-layout/README.md:# #require "compiler-libs"
```
Also, the "copy&paste" arguments doesn't really hold as you can't copy&paste the results.

In any case, what I meant is that we *must not* rely on ocamlfind in mdx for anything. This is just craft and will make it harder to remove in the feature if people start to use it.

Reply:
The issue here is that I don't think we can do without `findlib` atm.

A better and more future proof solution is discussed [here](https://github.com/ocaml/dune/issues/2470) and it's not clear whether it implies relying on `findlib` or not yet.

Anyway this is just here to get working tests in RWO. I'd understand if you don't want it merged because of it's fragile nature, I just wanted to avoid having to backport fixes or new features to a fork/branch. We can keep it separate for now.

Reply:
> The issue here is that I don't think we can do without findlib atm.

I am discussing about the API we expose to the user ; I am fine to use findlib internally at the moment, but again, we should not rely on it being present in the near future. Putting it in the block header makes it easier to switch to a different system (directly provided by dune) in the near future.

I am fine to merge that PR to ease maintenance, but before this I'd like to make sure we are not explicitely dependending on findlib/ocamlfind or at least these parts have to be clearly identified and separated so that they can be removed easily in a subsequent PR (which should happen before the next release).